src/back/spv/mod.rs

```diff
@@ -495,6 +495,11 @@ struct GlobalVariable {
     /// to refer to the global in the function body. This is the id of that access,
     /// updated for each function in `write_function`.
     access_id: Word,
+
+    /// The ID of a type that is the base for the array of bindings.
+    /// This is needed because the block encoder isn't expected to generate new types,
+    /// and there are cases where we need to get a pointer to one of those bases.
+    pointer_to_binding_array_element_type_id: Word,
```

Comment:
Why can't the block encoder just call `get_type_id` and pass a `LookupType::Local(LocalType::PointerToBindingArray { .. })`? These `GlobalVariable` handles are really confusing, so I'd like to avoid accumulating more stuff here if we can help it.

Reply:
I see now that `PointerToBindingArray` isn't what's needed here, it's a pointer to an element of the binding array. But it still seems like there should always be a `Type` in the arena for that, since it must have been declared as a global var of type `BindingArray`, which refers to its `base` via a handle.

Reply:
I believe you want the type of the inner `AccessIndex` expression's `base_outer` expression.

Reply:
I tried to express this in the comment - "This is needed because the block encoder isn't expected to generate new types". I.e. architecturally we have the block encoder assuming that all types that would be needed are already there.
I don't want to change that architectural restriction, I think there is elegance to it.
So the code in the PR does this type creation ahead of time, and it needs a place to cache those results.
This is in line with existing code, I don't think we should require this PR to re-engineer this aspect.